pkg/assert/validate.go

```diff
@@ -60,27 +54,29 @@ func runAssertion(
 	exec func(string) (string, error),
 	ticker <-chan time.Time,
 	timeout <-chan time.Time,
-	errorsChan chan<- error,
 ) error {
-	var res string
-	var err error
+	var (
+		res   string
+		err   error
+		retry int
+	)
+
 	for {
 		select {
 		case <-timeout:
 			timeoutErr := shared.ReturnLogError("timeout reached for command:\n%s\n"+
 				"Trying to assert with:\n%s\nExpected value: %s\n", cmd, res, assert)
-			errorsChan <- timeoutErr
 
 			return timeoutErr
 		case <-ticker:
-			i := 0
+			retry++
```

Comment:
assuming that retry need not be initialized here to zero - that ticker somehow knows the var takes care of it?

Reply:
the ticker actually has nothing to do with it....

we are just increment this var in case we get back on this select ( which in this case is a ticker )

Reply:
maybe i should explain better

one thing is the  ticker period of time 
another is the retry inside the err != nil 

thats why would make no sense have a big retry, since the timeout would finish first 

if we are still trying inside the ticker times it run , and still getting error, then we use the incremented var to break the loop some time ( after 5 retries ) 

if no error on next iteration we just increment the var , do nothing with it and exit the select/loop 

if no error never, still creating a var and again not using it and exiting the select/loop